### PR TITLE
bst.c : Added Binary Search Tree Implemetation [C]

### LINKED ISSUE #53: [Data Structure] Binary Search Tree [C]

### PR DESCRIPTION
**Pull request by Om-Pandey**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #53 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

**Comment by Om-Pandey** [comment]
Commits Squashed tets passed

File: insertion_sort/Java/insertionsort.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package java;
```
Comment (sangamcse) [inline insertion_sort/Java/insertionsort.java]: Is this BST? Please remove it from this PR.

File: insertion_sort/java/insertionsort.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package java;
```
Comment (sangamcse) [inline insertion_sort/java/insertionsort.java]: this too

**Comment by Om-Pandey** [comment]
yeah ... done ... sorry  .. actually continued commits from last PR I made on insertion sort

**Comment by sangamcse** [comment]
Update the README table also and Squash your commits into one.

For squash help, go through http://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits

**Comment by Om-Pandey** [comment]
Yeah ... about that .. how to update the README ?

**Comment by sangamcse** [comment]
@sangamcse rebase

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by sangamcse** [comment]
Open README in your text editor, update it according to given table.
> Yeah ... about that .. how to update the README ?

**Comment by sangamcse** [comment]
@Om-Pandey Rebase your local repo
Get help from https://github.com/NITSkmOS/Algorithms/blob/master/GIT_BASICS.md#follow-up
And ping me on gitter https://gitter.im/NITSkmOS/algo if I forget

**Comment by sangamcse** [comment]
Closing due to inactivity

**Pull request closed without merging**